src/ProjectTemplates/Microsoft.Extensions.AI.Templates/src/ChatWithCustomData/ChatWithCustomData.Web/NuGet.config.in

```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!--
+  This file only exists to allow locally-built packages to be used when running the template locally.
+  It will not get included in the built project template.
+-->
+<configuration>
+  <packageSources>
+    <add key="local-shipping" value="${ArtifactsShippingPackagesDir}" />
+  </packageSources>
+  <packageSourceMapping>
+    <packageSource key="local-shipping">
+      <package pattern="*" />
+    </packageSource>
+  </packageSourceMapping>
```

Comment:
This is required in order for the locally generated packages to get picked up during restore. I'm not 100% sure of the reason, but I think it's because the `<packageSourceMapping>` declared in the repo's root `NuGet.config` resulted in another package source getting used to resolve the `Microsoft.Extensions.AI` package reference, even though that source doesn't have the correct version.

If the `<packageSourceMapping>` element weren't required here, we could have removed this entire file and instead just modified `<RestoreAdditionalProjectSources>` in the `Directory.Build.targets.in` file.

Reply:
No longer needed. Resolving.